docs/guides/tfhers.md

```diff
@@ -0,0 +1,340 @@
+2. We will use the secret key meant for TFHE-rs to launch a partial keygen (i.e. starting from a secret key and generating the rest of the keys) in TFHE-rs
+
+Case 2.2 is basically the opposite, the steps are as follows:
+
+1. We do a classical keygen in TFHE-rs, which generates a single secret key and different public keys
+2. We will use the secret key to launch a partial keygen in Concrete
+
+It is possible to generate keys in both directions, meaning that you can start by generating keys from Concrete and then use Concrete's secret key to generate the remaining keys (partial keygen) in TFHE-rs (use case 1.1), or vice versa (use case 1.2).
+
+After the key generation is complete and both keysets are acquired, we can perform computations, encryption, and decryption on both ends.
+
+![alt text](../_static/tfhers/tfhers_uc1.excalidraw.svg)
+
+### Initial Setup and Config
+
+The first step is to define the TFHE-rs ciphertext we want to use (see [The Basics](#the-basics)). This includes specifying cryptographic and encoding parameters. We don't want to choose these parameters ourselves, because TFHE-rs already has a precomputed list of parameters we can use. We will be using [these ones](https://github.com/zama-ai/tfhe-rs/blob/2bd9f7aab412c92d0642a35fa87d1d0f4e2193b9/tfhe/src/shortint/parameters/classic/gaussian/p_fail_2_minus_64/ks_pbs.rs#L255) throughout this guide.
+
+In short, we will first look at what are good set of parameters we want to use in TFHE-rs, then just copy paste those in Concrete so that we have compatible ciphertexts (using the same parameters).
+
+```python
+from functools import partial
+from concrete.fhe import tfhers
+
+tfhers_params = tfhers.CryptoParams(
+    lwe_dimension=909,
+    glwe_dimension=1,
+    polynomial_size=4096,
+    pbs_base_log=15,
+    pbs_level=2,
+    lwe_noise_distribution=0,  # this won't be needed so we set it to 0
+    glwe_noise_distribution=2.168404344971009e-19,
+    encryption_key_choice=tfhers.EncryptionKeyChoice.BIG,
+)
+# creating a TFHE-rs ciphertext type with crypto and encoding params
+tfhers_type = tfhers.TFHERSIntegerType(
+    is_signed=False,
+    bit_width=8,
+    carry_width=3,
+    msg_width=2,
+    params=tfhers_params,
+)
+# this partial will help us create TFHERSInteger with the given type instead of calling
+# tfhers.TFHERSInteger(tfhers_type, value) every time
+tfhers_int = partial(tfhers.TFHERSInteger, tfhers_type)
+```
+
+### Defining the Circuit and Compiling
+
+We will now define a simple addition function. This function takes TFHE-rs inputs, converts them to Concrete format (`to_native`), runs a computation, and then converts them back to TFHE-rs. The circuit below is a common example that takes and produces TFHE-rs ciphertexts. However, there are other scenarios where you might not convert back to TFHE-rs, or you might convert to a different type than the input. Another possibility is to take one native ciphertext and one TFHE-rs ciphertext.
```

Comment:
We will now define a simple addition function -> We will now define a simple modular addition function